### PR TITLE
Query ace styles

### PR DESCRIPTION
**Pull request by kyleknighted**
## Description

- Updating variant colors that reverted at some point from https://github.com/Graylog2/graylog2-server/pull/7971
- Fixing Search Query input highlight color
- Updating positioning of Query Input to center with search button

## Motivation and Context

Needed more shades of variants and somehow I managed to revert previous changes to the variants. Also, the current updates to the Search Query input made highlighted text difficult to read.

## Screenshots (if appropriate):
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/searchbar/queryinput/StyledAceEditor.jsx [hunk 1]
```diff
@@ -8,9 +8,9 @@ const StyledAceEditor = styled(AceEditor).attrs(({ aceTheme, theme }) => ({
   theme: aceTheme,
   scTheme: theme,
 }))(({ scTheme }) => css`
-  margin-top: 9px;
-  height: 34px !important;
-  width: 100% !important;
+  height: 34px;
+  line-height: 32px;
+  width: 100%;
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/searchbar/queryinput/StyledAceEditor.jsx:13]: The query input has a fixed width, without `!important`:
image

File: graylog2-web-interface/src/theme/variants/teinte.js [hunk 2]
```diff
@@ -30,14 +30,24 @@ const variant = {
...
-  variantLight[name] = chroma(variant[name]).brighten(1).hex();
-  variantDark[name] = chroma(variant[name]).darken(1.5).hex();
+  if (typeof variant[name] === 'string') {
+    variant.light[name] = chroma.mix(variant[name], '#fff', 0.15).hex();
+    variant.lighter[name] = chroma.mix(variant[name], '#fff', 0.5).hex();
+    variant.lightest[name] = chroma.mix(variant[name], '#fff', 0.85).hex();
+
+    variant.dark[name] = chroma.mix(variant[name], '#000', 0.15).hex();
+    variant.darker[name] = chroma.mix(variant[name], '#000', 0.5).hex();
+    variant.darkest[name] = chroma.mix(variant[name], '#000', 0.85).hex();
```
Comment (linuspahl) [inline graylog2-web-interface/src/theme/variants/teinte.js:49]: You could create `lighten` for the top and a `darken` function for the bottom block.

**Comment by linuspahl** [comment]
@kyleknighted One thing I missed during the first review: The text cursor seems to be huge compared with the text size. Did you implemented this on purpose or mainly to centre the text?
I prefer using the old line-height an adding the margin inside the `QueryInput`.

**Comment by kyleknighted** [comment]
@linus the problem with margin is that long queries look bad
image

I will work on finding a solution

**Comment by kyleknighted** [comment]
@linuspahl pushed up a potential fix
image

File: graylog2-web-interface/src/views/components/searchbar/queryinput/StyledAceEditor.jsx [hunk 3]
```diff
@@ -8,140 +8,140 @@ const StyledAceEditor = styled(AceEditor).attrs(({ aceTheme, theme }) => ({
...
+    .ace_content,
+    .ace_placeholder {
+      top: 6px;
+      font-size: 16px;
+      padding: 0 !important;
+      font-family: inherit !important;
+    }
   }
 
   .ace_editor.ace_autocomplete {
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/searchbar/queryinput/StyledAceEditor.jsx:147]: @kyleknighted I like your changes regarding the highlight! While testing these changes, I noticed two problems with the autocomplete:

Due to the new changes the autocomplete lost its top "margin":
image
Compared with before (3.3 beta1):
image

And as you can see in the screenshot the autocomplete has a larger width. While having a look at the described problem, I saw the `  .ace_editor.ace_autocomplete {` styling is no longer being included. This is not directly related to this PR, but could you have a look at it as well?

Comment (kyleknighted) [reply graylog2-web-interface/src/views/components/searchbar/queryinput/StyledAceEditor.jsx:147]: got it fixed up

image

**Pull request merged**